File: frontend/src/types/exercise.ts
export type ExerciseCategory = 'STRENGTH' | 'CARDIO' | 'MOBILITY' | 'FUNCTIONAL'

export type MuscleGroup =
  | 'CHEST'
  | 'BACK'
  | 'LEGS'
  | 'SHOULDERS'
  | 'ARMS'
  | 'CORE'
  | 'CARDIO'
  | 'FULL_BODY'

export type ExerciseLevel = 'BEGINNER' | 'INTERMEDIATE' | 'ADVANCED'

export interface Exercise {
  id: string
  name: string
  description?: string
  category: ExerciseCategory
  muscleGroup: MuscleGroup
  level: ExerciseLevel
  videoUrl?: string
  imageUrl?: string
  equipment?: string
  createdAt: string
}

export interface ExercisePayload {
  name: string
  description?: string
  category: ExerciseCategory
  muscleGroup: MuscleGroup
  level: ExerciseLevel
  videoUrl?: string
  imageUrl?: string
  equipment?: string
}

File: backend/src/main/java/com/gymflow/pro/entity/Exercise.java
package com.gymflow.pro.entity;

import com.gymflow.pro.entity.enums.ExerciseCategory;
import com.gymflow.pro.entity.enums.ExerciseLevel;
import com.gymflow.pro.entity.enums.MuscleGroup;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "exercises")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Exercise {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "video_url")
    private String videoUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private ExerciseCategory category;

    @Enumerated(EnumType.STRING)
    @Column(name = "muscle_group", nullable = false, length = 50)
    private MuscleGroup muscleGroup;

    private String equipment;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private ExerciseLevel level = ExerciseLevel.BEGINNER;

    @Column(columnDefinition = "TEXT")
    private String instructions;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
